atst/models/mixins/auditable.py

```
@@ -95,6 +92,10 @@ def resource_type(self):
     def portfolio_id(self):
         return None
 
+    @property
+    def application_id(self):
+        return None
```

Can you see what happens if we `raise NotImplementedError()` here and in `portfolio_id` instead of returning `None`? This forces every model that implements the mixin to define those properties, otherwise things break. The advantage to this is that we'd catch the issues you found right away. The disadvantage is that you have to define them in every model and still `return None` in some cases. If it's not a huge burden to this PR, let's go for it. If so, can you make a follow-up PR tackling that?